docs/developer/develop-local.md

```diff
@@ -0,0 +1,47 @@
+# Developing on a local Turbinia setup (no cloud required)
+
+See [here](../../docker/local/README.md) on how to setup the local Turbinia stack with Docker.
+
+After you have the local stack up and running a usual development cycle would look like below.
+
+## Before you start
+*   Check out the [How it Works](../user/how-it-works.md) page to see how the different
+    components work within Turbinia.
+*   Make sure to follow the Turbinia
+    [developer contribution guide](contributing.md).
+
+#### Step 1
+Fork Turbinia on github and create a new feature branch to work on.
+```
+$ git clone https://github.com/[your-github-user-id]/turbinia.git
+$ git checkout -b my-new-feature
+$ cd turbinia
+```
+#### Step 2
+Add some awesome new feature, maybe [develop a new task](./developing-new-tasks.md)?
+#### Step 3
+Rebuild Turbinia server and/or worker Docker images.
```

Comment:
Just curious, how long does this take now?

Reply:
If you *only* update the turbinia code (and not requirements.txt) on my mac it takes 8 seconds for a rebuild of the image. It needs to change the last few layers (copy turbinia code in and setup.py install). As all requirements are already there it's super fast.

Reply:
Awesome.